### PR TITLE
feat(workflow): Assign alert to team on alert rules page

### PR DESCRIPTION
**Pull request by mikellykels**
Previously, if an alert rule wasn't assigned to a team, it would display `-` under Team. This PR adds a dropdown to assign a team to an alert rule from the alert rules page.

[FIXES WOR-741](https://getsentry.atlassian.net/browse/WOR-741)

# Before
Screen Shot 2022-05-18 at 1 28 27 AM

# After
Screen Shot 2022-05-18 at 12 44 39 AM

**Dropdown**
- Only displays the teams that are part of that project
Screen Shot 2022-05-18 at 12 47 45 AM

**After assigning**
- Can still update the assignee until page refresh
Screen Shot 2022-05-18 at 12 47 58 AM

- After page refresh
- Will need to "edit" the rule if you need to reassign
Screen Shot 2022-05-18 at 12 48 12 AM

**Disable if user doesn't have access to edit alerts**
- Also remove chevron
Screen Shot 2022-05-18 at 2 18 34 AM

**Success/Error messages**

Screen Shot 2022-05-18 at 1 11 15 AM
Screen Shot 2022-05-18 at 1 23 38 AM

### CODE REVIEW

File: static/app/views/alerts/list/rules/row.tsx [hunk 1]
```diff
@@ -55,6 +65,7 @@ function RuleListRow({
   userTeams,
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/row.tsx:65]: whats the difference between userTeams and filteredTeams?

File: static/app/views/alerts/list/rules/row.tsx [hunk 2]
```diff
@@ -221,6 +234,100 @@ function RuleListRow({
...
+            <Label>
+              <Highlight text={inputValue}>{`#${team.slug}`}</Highlight>
+            </Label>
+          </MenuItemWrapper>
+        ),
+      }))
+      .concat(unassignedOption);
+
+  const teamId = assignee && assignee.split(':')[1];
+  const teamName = filteredTeams.filter(team => team.id === teamId);
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/row.tsx:301]: .find

File: static/app/views/alerts/list/rules/row.tsx [hunk 3]
```diff
@@ -260,7 +367,47 @@ function RuleListRow({
...
+          <ActorAvatar actor={teamActor} size={24} />
+        ) : (
+          <AssigneeWrapper>
+            {!projectsLoaded && (
+              <LoadingIndicator
+                mini
+                style={{height: '24px', margin: 0, marginRight: 11}}
+              />
+            )}
+            <Access access={['alerts:write']}>
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/row.tsx:380]: maybe we can do this check once

File: static/app/views/alerts/list/rules/row.tsx (hunk 1 above)
Comment (mikellykels) [reply static/app/views/alerts/list/rules/row.tsx:65]: Ah my mistake - no difference, I was trying another approach earlier and forgot `userTeams` had what I needed. Updating it back to use `userTeams`

File: static/app/views/alerts/list/rules/index.tsx [hunk 4]
```diff
@@ -89,6 +89,29 @@ class AlertRulesList extends AsyncComponent<Props, State & AsyncComponent['state
     });
   };
 
+  handleOwnerChange = (
+    projectId: string,
+    rule: CombinedMetricIssueAlerts,
+    ownerValue: string
+  ) => {
+    const {params} = this.props;
+    const {orgId} = params;
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/index.tsx:98]: nit 
```suggestion
    const {orgId} = this.props.params;
```

File: static/app/views/alerts/list/rules/index.tsx [hunk 5]
```diff
@@ -89,6 +89,29 @@ class AlertRulesList extends AsyncComponent<Props, State & AsyncComponent['state
...
+      },
+      error: () => {
+        addMessage(t('Unable to save change'), 'error');
+      },
+    });
+  };
+
   handleDeleteRule = async (projectId: string, rule: CombinedMetricIssueAlerts) => {
     const {params} = this.props;
     const {orgId} = params;
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/index.tsx:117]: oh lol its again right here

File: static/app/views/alerts/list/rules/row.tsx [hunk 6]
```diff
@@ -357,4 +484,62 @@ const ActionsRow = styled(FlexCenter)`
...
+const MenuItemWrapper = styled('div')<{
+  disabled?: boolean;
+  py?: number;
+}>`
+  cursor: ${p => (p.disabled ? 'not-allowed' : 'pointer')};
+  display: flex;
+  align-items: center;
+  font-size: 13px;
+  ${p =>
+    typeof p.py !== 'undefined' &&
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/row.tsx:534]: can we remove this i don't think we need it

File: static/app/views/alerts/list/rules/row.tsx [hunk 7]
```diff
@@ -221,6 +237,80 @@ function RuleListRow({
...
+    disabled: false,
+  };
+
+  const projectRow = projects.filter(project => project.slug === slug);
+  const projectRowTeams = projectRow[0].teams;
+  const filteredProjectTeams =
+    projectRowTeams &&
+    projectRowTeams.filter(projTeam => {
+      return userTeams.has(projTeam.id);
+    });
```
Comment (scttcper) [inline static/app/views/alerts/list/rules/row.tsx:265]: nit
```suggestion
  const filteredProjectTeams = projectRowTeams?.filter(projTeam => {
      return userTeams.has(projTeam.id);
   });
```

File: tests/js/spec/views/alerts/rules/index.spec.jsx [hunk 8]
```diff
@@ -116,6 +122,44 @@ describe('AlertRulesList', () => {
     expect(rulesMock).toHaveBeenCalledTimes(0);
   });
 
+  it('displays team dropdown context if unassigned', async () => {
+    createWrapper();
+    const assignee = (await screen.findAllByTestId('alert-row-assignee'))[0];
+    const btn = (await within(assignee).findAllByRole('button'))[0];
```
Comment (scttcper) [inline tests/js/spec/views/alerts/rules/index.spec.jsx:128]: i think you should be able to use getAllByRole here

File: tests/js/spec/views/alerts/rules/index.spec.jsx [hunk 9]
```diff
@@ -116,6 +122,44 @@ describe('AlertRulesList', () => {
...
+
+  it('assigns rule to team from unassigned', async () => {
+    const assignMock = MockApiClient.addMockResponse({
+      method: 'PUT',
+      url: '/projects/org-slug/earth/rules/123/',
+      body: [],
+    });
+    createWrapper();
+    const assignee = (await screen.findAllByTestId('alert-row-assignee'))[0];
+    const btn = (await within(assignee).findAllByRole('button'))[0];
```
Comment (scttcper) [inline tests/js/spec/views/alerts/rules/index.spec.jsx:147]: same here

**Review by scttcper — APPROVED** [review]
just a bunch of nit picking on the `&&` syntax instead of `?.`

**Pull request merged**